pkg/osquery/tables/dataflattentable/helpers.go

```diff
@@ -40,6 +40,7 @@ func Columns(additional ...table.ColumnDefinition) []table.ColumnDefinition {
 		table.TextColumn("key"),
 		table.TextColumn("value"),
 		table.TextColumn("query"),
+		table.TextColumn("error"),
```

Comment:
I find myself unsure of `error` or `_error`. If I think about tables in general, I like `_error` to indicate something internal. But this is all datafatten land. No one is working with these columns directly, so `error` feels clean.

Reply:
On a semi-related note: as I thought about it more, I dislike the idea of adding an 'error' column to a DB table, but as you said this is not a typical sql table, so I think it's fine.

Reply:
I'm curious about what you're thinking.

Reply:
I just took a step back and imagined adding an 'error' column to a 'real' database table, and populating that column with internal error messages. It would feel.. weird. But, this isn't a 'real' database table, so I don't think it's an issue.

Reply:
It's not so obviously wrong to me. It bumps into all the in-band vs out-of-band signaling questions. We have some similar ones in jsonrpc. 

I think the core of the confusion is osquery looks like a database, but it really isn't. It's an API translation layer. There is no pile of data on disk that sqlite is querying. Instead, we use sqlite as a magic api translator. So we need some way to embed errors. 

Arguably it's better to do it via the statuses. But, because of the rest of the system, that just gets weird around being able to batch things everywhere.

I dunno, maybe I'm wrong. Maybe it's better to return a status error, instead of an error column

Reply:
But I also think there's no real correct answer, and we probably won't understand all the tradeoffs till we've worked with it for a little